core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -196,14 +196,31 @@ object ConsoleConsumer extends Logging {
 
     props.putAll(config.consumerProps)
     props.putAll(config.extraConsumerProps)
-    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, if (config.options.has(config.resetBeginningOpt)) "earliest" else "latest")
+    setAutoOffsetResetValue(config, props)
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.bootstrapServer)
     props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
     props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
     props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, config.isolationLevel)
     props
   }
 
+  def setAutoOffsetResetValue(config: ConsumerConfig, props: Properties) {
+    if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
+      // auto.offset.reset parameter was specified on the command line
+      if (config.options.has(config.resetBeginningOpt) && "latest".equals(props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG))) {
+        // conflicting options - latest und earliest, throw an error
+        System.err.println("Can't simultaneously specify --from-beginning and 'auto.offset.reset=latest', please remove one option")
+        Exit.exit(1)
```

Comment:
I'm guessing we don't have a test case for this because of the system exit? However, it looks like the `Exit` object does allow us to override the behavior with `Exit.setExitProcedure`. Maybe we could use that in a test case? Just need to be careful to reset the state with `resetExitProcedure`.